src/series25.rs

```diff
@@ -158,3 +164,84 @@ impl<SPI: Transfer<u8>, CS: OutputPin> Flash<SPI, CS> {
         spi_result.map(|_| ()).map_err(Error::Spi)
     }
 }
+
+impl<SPI: Transfer<u8>, CS: OutputPin> BlockDevice<u32, SPI, CS> for Flash<SPI, CS> {
+    const SECTOR_LENGTH: usize = 4096;
+
+    unsafe fn erase_bytes_unchecked(
+        &mut self,
+        addr: u32,
+        amount: usize,
+    ) -> Result<(), Error<SPI, CS>> {
+        let amount = amount / Self::SECTOR_LENGTH;
+        for c in 0..amount {
+            self.write_enable()?;
+
+            let current_addr: u32 = (addr as usize + c * 256).try_into().unwrap();
+            let mut cmd_buf = [
+                Opcode::SectorErase as u8,
+                (current_addr >> 16) as u8,
+                (current_addr >> 8) as u8,
+                current_addr as u8,
+            ];
+            self.command(&mut cmd_buf)?;
+
+            let mut done = false;
+            // Wait until the erase is done
+            // TODO: maybe exchange this with a delay
+            while !done {
+                let status = self.read_status()?;
+                done = (status & Status::BUSY).is_empty();
+            }
```

Comment:
This loop could be moved into a utility function to reduce code duplication.

I also think it could be simplified a bit by doing this:

```rust
while self.read_status()?.contains(Status::BUSY) {}
```